### ISSUE TITLE
Cannot wake from sleep after enabling "virtual machine platform" in AMD laptop

### ISSUE THREAD

**Issue by hzxa21** [microsoft/WSL#5690]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.19041.388]
Your Distribution version: Ubuntu
Whether the issue is on WSL 2 and/or WSL 1: WSL2
```

# Steps to reproduce

- Turn on "virtual machine platform" windows feature
- Go to start menu, choose power -> sleep
- After ~10s, try to wake up the laptop by keyboard/touchpad/power button.

#  Expected behavior
Keyboard/touchpad/power button can wake my laptop up.

# Actual behavior
Nothing can wake my laptop up from sleep mode. keyboard/touchpad/power button won't work.

# Notes
- My laptop has the AMD Ryzen 4700U CPU. I saw online that many people with a AMD laptop complains about the same issue and someone mentions that there is a software problem in windows10 controller component.
- If I turn off "virtual machine platform", I can wake my laptop from sleep mode without any issue
- I like WSL2 very much and I want to use it at a daily-basis for development. Please help fix the issue!

**Comment by therealkenc** [comment]
Same #5549. Yours can be the new LZ.

**Comment by hcen1997** [comment]
hi therealkenc i heve this issue too  
in my amd laptop, if i try to turn on 'virtual machine platform' and restart windows.  
it will show lenovo logo as uefi logo and don't show loading circle and stay black screen with high cpu usage (based on fan)  
after l enter safety mode, the system said 'Unable to complete the feature configuration, the changes are being undone' then back to normal without 'vitual machine platform'  

i think there is some thing wrong with my bitlocker.(https://github.com/microsoft/WSL/issues/4103  

so i will unlock volume and wait

**Comment by hcen1997** [comment]
no not caused by bitlocker.  
still go black and see  'Unable to complete the feature configuration, the changes are being undone'

**Comment by hzxa21** [comment]
I tried turning off bitlocker on my laptop but it still cannot wake from sleep if "Virtual Machine Platform" is turned on.

**Comment by Darkein** [comment]
I have the same problem with a redmibook powered by ryzen and a fresh install of windows 10 pro 2004. Everything worked correctly until I install wsl 2 :/

**Comment by Darkein** [comment]
just reinstalled windows and it worked again. I tried to install WSL2 step by step and this command broke sleep mode:
'dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart'
I don't know if it's the best place to talk about this bug now :/

**Comment by hzxa21** [comment]
@Darkein Yeah, I also noticed that the wake up from sleep won't work if virtual machine platform is enabled. Hope this will be fixed soon.

**Comment by 2001zhaozhao** [comment]
Same issue on my HP envy x360 13 (4700U) and my desktop (3700x). Both of them can not sleep properly with virtualization on but hibernate works fine.

**Comment by motherfucka** [comment]
I have the same problem...
AMD Phenom II X4 955

**Comment by hzxa21** [comment]
Upgraded to Microsoft Windows [Version 10.0.19041.450]. The same issue still persists.

**Comment by jeff84** [comment]
I have the same issue with my HP Envy x360 13-ay0285ng with AMD Ryzen 7 4700u.

**Comment by esskay** [comment]
Same issue with HP ENVY x360 Laptop - 13-ay0045au with AMD Ryzen 4500U
Using slower WSL1 for now which does not support mounting of images :(

**Comment by Tarik02** [comment]
Acer Swift 3 SF314-42 the same issue. Models with both Ryzen 4500U and Ryzen 4700U.

**Comment by markspilot** [comment]
Same here with HP ENVY x360  - 13-ay0504sa  Ryzen 4500U running WSL2

**Comment by nivagator** [comment]
Same here with HP Envy x360 13-ay0075nr Ryzen 4700U. I am also forced to use WSL1 and a remote machine until this is resolved.

**Comment by markspilot** [comment]
> Same here with HP Envy x360 13-ay0075nr Ryzen 4700U. I am also forced to use WSL1 and a remote machine until this is resolved.

How to I downgrade to WSL1?

**Comment by nivagator** [comment]
> > Same here with HP Envy x360 13-ay0075nr Ryzen 4700U. I am also forced to use WSL1 and a remote machine until this is resolved.
> 
> How to I downgrade to WSL1?

```
wsl --set-version <Distro> 1
```

**Comment by markspilot** [comment]
> > > Same here with HP Envy x360 13-ay0075nr Ryzen 4700U. I am also forced to use WSL1 and a remote machine until this is resolved.
> > 
> > 
> > How to I downgrade to WSL1?
> 
> ```
> wsl --set-version <Distro> 1
> ```

Do I need to disable WSL2 or somthing?

**Comment by showyou** [comment]
Same issue with HP Envy x360 13-ay0xxxx Ryzen 4700U. I am forced to use WSL1 and cannot use Docker.

**Comment by dapao9999** [comment]
Same here 4600u lenovo xiaoxin air14

**Comment by MU-Software** [comment]
Happened on HANSUNG(not Samsung!) TFX5470H Laptop. It uses Ryzen 4800H.
Had to open laptop backplate and unplug battery terminal😥

**Comment by alfuhigi** [comment]
Same issue with HP ENVY x360 Laptop - 13-ay0001au with AMD Ryzen 4700u
 WSL2

**Comment by farhanotahmid** [comment]
Same issue on my HP ENVY x360 Convertible 15m-ee0xxx AMD Ryzen 5 4500U. I really want to use WSL2 but I have to miss out because of this error.

**Comment by xpierrohk** [comment]
On my 4500U, it made the screen flickers at regular interval. There's a definite issue with WSL2 / VM Platform on these CPUs !

**Comment by Tarik02** [comment]
@xpierrohk that’s an issue with virtualisation in particular. It disappears if you disable virtualisation in bios and appears to happen both in Windows and Linux.

**Comment by allumik** [comment]
Can confirm this also, HP Envy x360 here with Ryzen 4700. Can't do without wsl2, so i temporarily solved it by going to hibernate instead of sleep all the time. Is there some forum post/issue tracker thingy that addresses this on the AMD side?

**Comment by Tarik02** [comment]
> > @xpierrohk that’s an issue with virtualisation in particular. It disappears if you disable virtualisation in bios and appears to happen both in Windows and Linux.
> 
> 
> 
> Interesting, so would that mean the entire feature from AMD is fake and never actually works (well I mean, without blinking half the screen at random) ? Out of curiosity, if it happens to me on Linux, which equivalent to the "virtual machine platform" should I uninstall ?

I have no any problems with screen blinking, the only problem is sleep. It can be replaced with hibernation, but that's more temporary solution than general one. I think, we should wait for fix from amd.

**Comment by Tarik02** [comment]
> Can confirm this also, HP Envy x360 here with Ryzen 4700. Can't do without wsl2, so i temporarily solved it by going to hibernate instead of sleep all the time. Is there some forum post/issue tracker thingy that addresses this on the AMD side?

https://community.amd.com/thread/254849

**Comment by AstroJosePC** [comment]
Same issue on HP Envy x360 15m-ee0013dx with Ryzen 5 4500U. I can't use WSL2 until they fix this issue. 
Sleep mode works as expected with virtualization disabled.

**Comment by nimrodolev** [comment]
Experiencing the exact same issue with an HP Envy x360 Convertible 13-ay0xxx, can't use WSL2 (for docker-desktop) while this issue persists.
Subscribing and crossing my fingers this gets sorted 😢

**Comment by fjluengo** [comment]
Same issue with HP ENVY x360 Convertible 13-ag0xxx AMD Ryzen 5 2500U and WSL2 enabled.

**Comment by vprouza** [comment]
Same issue on ENVY 13-ay0001nc

**Comment by gellweiler** [comment]
I've just installed the latest BIOS update on my HP ENVY 15-ee0266ng via the HP Support Assistant. It looks like that finally fixed the issue and I can now use suspend again.

**Comment by mikepotter** [comment]
After installing the BIOS update, as gellweiler mentioned, waking from sleep is working on my ENVY x360.

**Comment by allumik** [comment]
Can confirm this is fixed with the new BIOS update (f.13) on HP ENVY x360 13-ay0xxx with Ryzen 4700u

**Comment by Tarik02** [comment]
Swift 3 SF314-42 got bios update but that didn’t fix the issue...

**Comment by showyou** [comment]
I can confirm this is fixed after installing the latest BIOS update (f.13) on my ENVY x360 13-ay0xxx.

**Comment by hochunleung** [comment]
> The same issue with my Lenovo Xiaoxin Air 14
> AMD Ryzen 4600U

It seems fixed after updating the BIOS to DTEC20WW(1.06) version.

**Comment by jdeloshoyos** [comment]
> Swift 3 SF314-42 got bios update but that didn’t fix the issue...

Can confirm. Swift 3 SF314-42 latest BIOS update as of today (v1.07) does *not* fix the issue.

I'm hopefulm though; it seems that a BIOS update can, in fact, fix it, so i'll have to wait for now...

**Comment by esskay** [comment]
> Same issue with HP ENVY x360 Laptop - 13-ay0045au with AMD Ryzen 4500U
> Using slower WSL1 for now which does not support mounting of images :(

updating bios to version F.13 resolved the issue

**Comment by woodywould** [comment]
Looks like the same issue for me too,  this time on a Dell Inspiron w/AMD Ryzen 4000 series.

**Comment by trey-robinson** [comment]
Same issue, Lenovo Thinkpad E15 Gen 2 with Ryzen 4500U.

**Comment by jdeloshoyos** [comment]
> Same issue, Lenovo Thinkpad E15 Gen 2 with Ryzen 4500U.

Is there any BIOS update available for your computer? That seems to have fixed the issue for some folks. But still, I'm confused as to what's the true cause of the issue: if it's a problem in the BIOS for devices with Ryzen CPUs, or a bug in Hyper-V/WSL, or even a bug in some AMD driver (graphics, etc.) on these systems. From what I can gather here and in other sites, the answer to that is not clear yet.

**Comment by trey-robinson** [comment]
> 
> 
> > Same issue, Lenovo Thinkpad E15 Gen 2 with Ryzen 4500U.
> 
> Is there any BIOS update available for your computer? That seems to have fixed the issue for some folks. But still, I'm confused as to what's the true cause of the issue: if it's a problem in the BIOS for devices with Ryzen CPUs, or a bug in Hyper-V/WSL, or even a bug in some AMD driver (graphics, etc.) on these systems. From what I can gather here and in other sites, the answer to that is not clear yet.

I have, actually. That was one of the first things I tried trying to resolve this issue, updating the BIOS.

And actually, I realized I had misread the thread. I was having the _opposite_ issue; that is, sleep mode stops working if I DISABLE hyper-v and virtual platform! From this link https://community.amd.com/thread/254849 I saw the suggestion to turn on Hyper-V and indeed, no more issues with sleep mode (so far).
Although I need my VBox VMs and they don't work with Hyper-V enabled so ultimately there's still a problem, haha.

**Comment by svengo** [comment]
> > Swift 3 SF314-42 got bios update but that didn’t fix the issue...
> 
> Can confirm. Swift 3 SF314-42 latest BIOS update as of today (v1.07) does _not_ fix the issue.

BIOS update version 1.08 from 16.11.2020 fixed this problem for me.

**Comment by Tarik02** [comment]
> > > Swift 3 SF314-42 got bios update but that didn’t fix the issue...
> > 
> > 
> > Can confirm. Swift 3 SF314-42 latest BIOS update as of today (v1.07) does _not_ fix the issue.
> 
> BIOS update version 1.08 from 16.11.2020 fixed this problem for me.

Did not fix for me though. Sadly...

**Comment by jdeloshoyos** [comment]
> > > Swift 3 SF314-42 got bios update but that didn’t fix the issue...
> > 
> > 
> > Can confirm. Swift 3 SF314-42 latest BIOS update as of today (v1.07) does _not_ fix the issue.
> 
> BIOS update version 1.08 from 16.11.2020 fixed this problem for me.

1.08 BIOS didn't fix it for me either! I'm at my wits' end about this... Can it be a driver which needs updating too? BIOS v1.08 only mentions a fix for a Bluetooth recovery issue in the changelog. All Bluetooth drivers seem to be up to date in my laptop...

@svengo , could you share a little bit more info about your setup (Windows version, etc.)? I'm on Windows 10 20H2, fully up to date in Windows Update. Did you reset your BIOS settings (I tried that, didn't work) or chose any specific setting? I'd very much like to pinpoint what really is the cause of the issue, especially if you could solve it.

**Comment by svengo** [comment]
BIOS update 1.0.8 for Swift 3 SF314-42...

> @svengo , could you share a little bit more info about your setup (Windows version, etc.)? I'm on Windows 10 20H2, fully up to date in Windows Update. Did you reset your BIOS settings (I tried that, didn't work) or chose any specific setting? I'd very much like to pinpoint what really is the cause of the issue, especially if you could solve it.

I am using the same Windows Version also with up to date drivers. But I was probably too hasty. Yesterday it seemed  to work, currently the laptop doesn't go into standby mode anymore -  but at least it doesn't crash anymore.  :-/

**Comment by dlguswo333** [comment]
For those who still have the problem,
I recommend you to update your graphics driver to **V27.20.11024.11003**.
I have updated the driver via windows 10 update, and it solved other issues as well as the sleep issue.
You can check the lenovo employee's comment at the link below:
> https://forums.lenovo.com/t5/ThinkPad-11e-Windows-13-E-and-Edge-series-Laptops/E15-Gen-2-Cannot-wake-from-sleep/m-p/5039906

**Comment by svengo** [comment]
@jdeloshoyos, @Tarik02, Update on Acer Swift 3 SF314-42:

I just got the answer from the german Acer support (incident: 201014-005087) that they were able to reproduce the problem and found a firmware error as cause. Acer would publish a corresponding update (V1.09) on their website shortly (approx. end of December 2020).

**Comment by Tarik02** [comment]
> @jdeloshoyos, @Tarik02, Update on Acer Swift 3 SF314-42:
> 
> 
> 
> I just got the answer from the german Acer support (incident: 201014-005087) that they were able to reproduce the problem and found a firmware error as cause. Acer would publish a corresponding update (V1.09) on their website shortly (approx. end of December 2020).

That's cool. Can't wait for the update.

**Comment by fcieminski** [comment]
On Lenovo Legion 5 with Ryzen 7 4800H the sleep mode works well, but there is a issue with screen ocasionally blinking. Definitelly because of virtual machine platform, there is no issue if I turn it off.  The bios update didn't fix it. Basically I can work with WSL2 only if I turn off the hybrid mode and run only on GTX 1650.

**Comment by motherfucka** [comment]
WOW. Last windows update + docker 3.0 update FIXED this problem

AMD Phenom II X4 955

**Comment by Animeshz** [comment]
Same issue, writing from Acer Swift 3 SF314-42. Recently turned on Hyper-V and Hypervisor platform along with installing WSL2+Docker, no updates whatsoever.

Can confirm everything was normal for 6 months, as soon as Hyper-V is on, issue comes.

**Comment by Animeshz** [comment]
By the way how are ya all living with this issue? Reverting the installation, disabling sleeps, or something else? I have currently no idea what to do right now :(

**Comment by pkjennings999** [comment]
@Animeshz I have the same model. Had to reinstall Windows. I was alright doing that as I only had the machine a day before I noticed the issue. I tried disabling the virtual machine platform and uninstalling WSL but the issue still persisted. I don't know of any other way to fix the issue

**Comment by Tarik02** [comment]
@Animeshz I have disabled sleep and I’m using hibernation instead of sleep. It is some slower (not much btw), but works good. And I binded Fn+F1 to hibernation.

**Comment by gpsietzema** [comment]
I have a ThinkPad T14 with a Ryzen 4750 Pro and have the same issue. Bios update doesn't help, changing 'Sleep' to 'Hybernate' does work for now, even though I want my 'Sleep' mode back ;-)

**Comment by svengo** [comment]
@jdeloshoyos, @Tarik02, @Animeshz:
I received a beta version of the BIOS V1.09.PI1001.P for my Swift SF314-42 from acer support, which solves the standby problem. According to support, however, this version would not have been tested conclusively and there would be a small risk associated with its use. 

The last ticket number at the German acer support is Incident: 210109-001663.

**Comment by Tarik02** [comment]
@svengo I'm very glad to hear that. Can't wait for that update.

**Comment by jdeloshoyos** [comment]
> @jdeloshoyos, @Tarik02, @Animeshz:
> I received a beta version of the BIOS V1.09.PI1001.P for my Swift SF314-42 from acer support, which solves the standby problem. According to support, however, this version would not have been tested conclusively and there would be a small risk associated with its use.
> 
> The last ticket number at the German acer support is Incident: 210109-001663.

Cool!! Did you try it out, or will wait for the official release?

At least, this shows they actually have been working on the issue. Sooner or later the stable, official version will come out. Thanks a lot for sharing this!

**Comment by aelfinn** [comment]
> @svengo 
> I received a beta version of the BIOS V1.09.PI1001.P for my Swift SF314-42 from acer support, which solves the standby problem. 

Do you think you could make that version available to others so we can see whether it solves the problem for us as well? That would be much appreciated!

**Comment by svengo** [comment]
> > @aelfinn 
> Do you think you could make that version available to others so we can see whether it solves the problem for us as well? That would be much appreciated!

I don't think I'm allowed to post the update. It is best to contact Acer support yourself. If there are enough bug reports, maybe Acer will speed up the release of the update.

**Comment by aelfinn** [comment]
> @svengo 
> I don't think I'm allowed to post the update. It is best to contact Acer support yourself.

I did. They told me they didn’t know of any beta version: “Ein "Beta" Bios V1.09 für Ihr Notebook ist uns nicht bekannt.” :D

Maybe you could send me a link via email?

**Comment by jdeloshoyos** [comment]
> > > @aelfinn
> > > Do you think you could make that version available to others so we can see whether it solves the problem for us as well? That would be much appreciated!
> 
> I don't think I'm allowed to post the update. It is best to contact Acer support yourself. If there are enough bug reports, maybe Acer will speed up the release of the update.

Did you try it out, anyway? Does it work?

**Comment by svengo** [comment]
> > I don't think I'm allowed to post the update. It is best to contact Acer support yourself. If there are enough bug reports, maybe Acer will speed up the release of the update.
> Did you try it out, anyway? Does it work?

Yes, I did not have any more crashes after the update. The "Connected Standby" seems to work normally. I could not find any negative side effects.

**Comment by jdeloshoyos** [comment]
> Yes, I did not have any more crashes after the update. The "Connected Standby" seems to work normally. I could not find any negative side effects.

It's been more than a month since this, and Acer still does not release the updated BIOS. It might be that they have found some blocking problem with the update, and of course, in the case of something as important and low-level as the BIOS, they must make sure the final product is rock-solid. Still, seeing as some other brands released updated and working BIOS regarding this same problem months ago, to be kept waiting is becoming more and more annoying as days pass ;-)

**Comment by Animeshz** [comment]
The acer's bios update just been released guys! Check it out, now I have everything working correctly!!!

**Comment by jdeloshoyos** [comment]
> The acer's bios update just been released guys! Check it out, now I have everything working correctly!!!

Yes, can confirm! New 1.09 BIOS for Acer Swift 3 (SF314-42) correctly resolves this issue. Now I can keep using Docker/WSL2 and have my computer sleep without problems.

**Comment by svengo** [comment]
> The acer's bios update just been released guys! Check it out, now I have everything working correctly!!!

Just for the record: This BIOS update from acer has the same version number as the beta version, which has been running smoothly on my system since January.

**Comment by aelfinn** [comment]
Updated my SF314-42 to official BIOS v.1.09; standby on Kubuntu 20.04.2 _not working_ just as before.

**Comment by Tarik02** [comment]
@aelfinn the problem is in sleep states (modern standby). Linux does not fully support this and can't enter sleep state (s0 with some letters). Before, sleep was implemented with s3 state. I recommend you downgrading your BIOS to v1.04 so you can access advanced menu (turn off laptop, click buttons F4, 4, r, f, v, F5, 5, t, g, b, F6, 6, y, h, n) then you can disable modern standby.

**Comment by aelfinn** [comment]
> @aelfinn the problem is in sleep states (modern standby). Linux does not fully support this and can't enter sleep state (s0 with some letters). Before, sleep was implemented with s3 state. I recommend you downgrading your BIOS to v1.04 so you can access advanced menu (turn off laptop, click buttons F4, 4, r, f, v, F5, 5, t, g, b, F6, 6, y, h, n) then you can disable modern standby.

Thing is: I can't downgrade to 1.04 — the BIOS updating software gives an error because the BIOS to be flashed isn't newer than the on installed. :(

**Comment by Tarik02** [comment]
> Thing is: I can't downgrade to 1.04 — the BIOS updating software gives an error because the BIOS to be flashed isn't newer than the on installed. :(

You can, but you have to extract update exe first and then edit config. I will describe this more detailed later.

**Comment by Tarik02** [comment]
> Thing is: I can't downgrade to 1.04 — the BIOS updating software gives an error because the BIOS to be flashed isn't newer than the on installed. :(

You can extract update exe file using 7z. There is `platform.ini` file and inside there's section `[Bios_Version_Check]`, inside `Flag` key. Set it to 0 and run `H2OFFT-Wx64.exe` (not sure about exe file). It should downgrade BIOS.

**Comment by muhammadyusuf-kurbonov** [comment]
Same issue with Lenovo Ideapad gaming 3. Intel Core-i5 10300H. Up to date software.

**Comment by muhammadyusuf-kurbonov** [comment]
Has someone same issue with Intel CPU?
I have same issue with Intel Core i5-10300H. Lenovo ideapad gaming 3

**Comment by Zelgadiss** [comment]
> Has someone same issue with Intel CPU?
> I have same issue with Intel Core i5-10300H. Lenovo ideapad gaming 3

I have the same issue on Intel desktop i7-7800x

**Comment by nikhilreddydev** [comment]
@Tarik02 have you got your SF314-42 sleep issue resolved with the BIOS v1.09 update?
I am  currently running on v1.04 on my SF314-42.

**Comment by Tarik02** [comment]
@nikhilreddydev, yes, BIOS 1.09 does not have such problem.

**Comment by nikhilreddydev** [comment]
I had same issues with wsl2 with ubuntu 20.04, my laptop cant able to wake up from sleep. so I disabled VMP.
Thanks for that fast reply tho.

**Comment by Tarik02** [comment]
> I had same issues with wsl2 with ubuntu 20.04, my laptop cant able to wake up from sleep. so I disabled VMP.
> 
> Thanks for that fast reply tho.
> 
> 

Actually, it fixed the issue with windows, not sure about Linux. As I know, Linux works very bad with modern standby and it can cause issues.

**Comment by nikhilreddydev** [comment]
Okay, I just need the wsl2 to be working on my machine without any sleep issues.
I got firmware update just now.

Cant wait for wsl2 + docker + windows working seamlessely

**Comment by hingashi-coder** [comment]
Same problem on ASUS ROG Zephyrus G14  Ryzen 9 4900HS.

**Comment by Harvtronix** [comment]
Same issue for me. When I install Docker (which installs WSL2 and Virtual Machine Platform), I immediately begin seeing sleep wakeup hangs. As soon as I uninstall it (and WSL2 and VMP), the sleep problems go away.

Hardware:

- Mobo: Gigabyte Z390M Gaming
- CPU: Core i7-9700k
- GPU: Radeon RX 6800 XT

**Comment by Animeshz** [comment]
For linux users, set CONFIG_AMD_PMC to m or y in the linux kernel's menuconfig before building the kernel (this is missing in some distros as a result s0ix/s2idle does not function properly), it should now work as expected, have tested on kernel v5.15.2 ([a bit more info](https://gitlab.freedesktop.org/drm/amd/-/issues/1721#note_1161826)).

**Comment by elzik** [comment]
I have had the same issue on my DELL Inspiron 7373 laptop for a long time now. I have been trying to find the cause for this for a long time. I now understand that it is installing Docker Desktop (and therefore Virtual Machine Platform) that breaks both Sleep and Hibernate for me and would cause the machine to stay on and heat-up. Within 6 months it had totally destroyed my battery.

Uninstalling Virtual Machine Platform fixes the problem (but obviously stops me from using Docker).

Have any DELL Inspiron users faced the same issue? Has anyone found a fix? I'd like to be able to use Docker again. I have the latest BIOS and drivers from DELL.

**Comment by hingashi-coder** [comment]
I tried to use Docker with Hyper-V(Without WSL2)
But It won't fix.

**Comment by muhammadyusuf-kurbonov** [comment]
It was fixed, when I reinstall OS with Secure boot, on GPT (earlier it was on MBR) and have installed all updates of that time (it was about a year ago).

**Comment by jamest-portainer** [comment]
I got the same issue with a Dell XPS 15 9500 laptop running Windows 11 Pro 21H2 after I had installed Docker Desktop, which also installs WSL2.

I found that the machine would hang (at the Dell logo) on attempting to wake from sleep.

Also, it would come back to that same hanging state on shutting the lid, presumably it went into sleep and then tried to wake again. This meant that the laptop would get extremely hot, with fans running full blast, etc.

I tried all sorts of settings to try to resolve but I couldn't and had to always either leave my machine on or shut it down. In the end I reinstalled Windows. I am not keen to reinstall WSL2 unless I know the issue is resolved.

**Comment by Ecdldaiiere** [comment]
Thank god for opening up this issue. One day out of the blue my Windows 10 Acer Swift 3 AMD computer would not wake up from sleep, and every time I had to hold down the power button to turn it off and then one more time to turn it back on again. I tried all the tutorials online concerning problems waking up computer from sleep but nothing worked. 

This went on for a few days, and I was seriously considering buying a new laptop because I thought it was something wrong with my batteries. I didn't know why this started suddenly happening. I only knew that it started happening on the day I installed Docker. But I didn't connect the dots until I finally searched "computer won't wake up if sleeping too long docker" and found this thread and turned off "Virtual Machine Platform". Now the issue is completely fixed. Thank you so much for making this thread and saving me from so much frustration and confusion.

**Comment by florian-leblanc** [comment]
I searched for weeks to find out why I couldn't put my computer into sleep mode. Finally, I discovered that it happened after I installed WSL2. When I unchecked the 'Virtual Machine Platform' option, the sleep mode started working again, but WSL2 stopped functioning.

Here is my configuration:
Motherboard: X670 AORUS ELITE AX
Processor: AMD Ryzen 9 7950X 16-Core

Windows version:
Edition: Windows 11 Pro N
Version: 22H2
Installed on: 20/05/2023
OS build: 22621.1702
Experience: Windows Feature Experience Pack 1000.22641.1000.0

**Comment by b1scu1t** [comment]
I have the same issue with the Windows Hypervisor/WSL2, but not consistently with a Dell Precision 7520. 

With WSL2/Hypervisor enabled, the laptop will wake from sleep normally almost every time (4-in-5) when the charger is connected. 

If the charger is disconnected/runs on battery and put to sleep, there is a 2-in-3 chance that the system fails to wake; the power button and keyboard lights-up, but no HDD activity and the screen remains dead.

Disabling the Hypervisor/WSL2 with `bcdedit /set hypervisorlaunchtype auto` solves the problem, but I lose WSL2.

I've tried updating the BIOS and switching between the Dell-provided drivers and the generic up-to-date ones from Nvidia and Intel...they don't work.
 
 Windows 10 Enterprise LTSC
 Kernel: 10.0.19044.0
 Host: Dell Inc. Precision 7520
 CPU: Intel(R) Core(TM) i7-6820HQ CPU @ 2.712GHz
 GPU: NVIDIA Quadro M2200
 GPU: Intel(R) HD Graphics 530
 
I genuinely thought the laptop was kicking the bucket for a long time...

**Comment by craigloewen-msft** [comment]
Hi folks, for issues like this they are more than likely firmware issues so please try and update your firmware as much as you can. If it still persists then please open a new issue with details for us to take a look at. I'm going to close this issue out for our bookkeeping.
